jax/_src/custom_partitioning.py

```diff
@@ -527,8 +527,16 @@ def _custom_partitioning_lowering_rule(ctx: mlir.LoweringRuleContext, *values,
   if isinstance(axis_context, sharding_impls.ShardingContext):
     devices = axis_context.device_assignment
     if devices is None:
-      raise AssertionError(
-          'Please file a bug at https://github.com/jax-ml/jax/issues')
+        raise AssertionError(
+            "You may be using a primitive in the context of a custom partitioning specialization "
+            "that requires devices during lowering. However, no device assignment was found. "
+            "To fix this issue, ensure that the primitive is added to:\n"
+            "    dispatch.prim_requires_devices_during_lowering\n"
```

Comment:
I don't think we want to put private APIs in the error message.

Now I know that's suboptimal but maybe we can document it in some other way in the docs somewhere

Reply:
It was mentioned in the changelog: https://github.com/jax-ml/jax/blob/main/CHANGELOG.md#jax-0424-feb-6-2024

maybe that's enough for now?